packages/flutter/test/widgets/transitions_test.dart

```diff
@@ -457,6 +518,66 @@ void main() {
     });
   });
 
+  group('MatrixTransition', () {
+    testWidgets('uses ImageFilter when provided with FilterQuality argument', (WidgetTester tester) async {
+      final AnimationController controller = AnimationController(vsync: const TestVSync());
+      final Animation<double> animation = Tween<double>(begin: 0.0, end: 1.0).animate(controller);
+      final Widget widget = Directionality(
+        textDirection: TextDirection.ltr,
+        child: MatrixTransition(
+          animation: animation,
+          onTransform: (double value) => Matrix4.identity(),
+          filterQuality: FilterQuality.none,
+          child: const Text('Matrix Transition'),
+        ),
+      );
+
+      await tester.pumpWidget(widget);
+
+      // Validate that expensive layer is not left in tree before animation has started.
+      expect(tester.layers, isNot(contains(isA<ImageFilterLayer>())));
+
+      controller.value = 0.25;
+      await tester.pump();
+
+      expect(
+          tester.layers,
+          contains(isA<ImageFilterLayer>().having(
+            (ImageFilterLayer layer) => layer.imageFilter.toString(),
+            'image filter',
+            startsWith('ImageFilter.matrix('),
+          )));
```

Comment:
formatting nit, use a trailing comma:

```suggestion
      expect(
        tester.layers,
        contains(isA<ImageFilterLayer>().having(
          (ImageFilterLayer layer) => layer.imageFilter.toString(),
          'image filter',
          startsWith('ImageFilter.matrix('),
        )),
      );
```

Reply:
(same for the two expects below)